lib/random.php

```diff
@@ -296,14 +273,15 @@ function random_int($min, $max)
              * Apply mask
              */
             $val &= $mask;
+            $val += $valueShift;
```

Comment:
So in the extended-range case, `$valueShift` is zero, and `$mask` is all ones, so `$val` will be a random integer between `PHP_INT_MIN` and `PHP_INT_MAX`. Since the range overflows an integer, we know it's at least `PHP_INT_MAX`, and therefore there's a ~50% probability that random integer `$val` will fall in the range. Is that the reasoning here?

That's actually pretty clever...